src/main/java/com/r307/arbitrader/service/TradingService.java

```diff
@@ -568,9 +615,62 @@ private void logEntryTrade(Spread spread, String shortExchangeName, String longE
 
     }
 
-    // get volume for an entry position considering exposure and exchange step size if there is one
-    private BigDecimal getVolumeForEntryPosition(BigDecimal maxExposure, BigDecimal price, int scale) {
-        return maxExposure.divide(price, scale, RoundingMode.HALF_EVEN);
+    /**
+     * Calculates the volume to trade on the long exchange such as:
+     * - the total price does not exceed the long exchange maximum exposure
+     * - the total price to trade on the short exchange does not exceed the short exchange maximum exposure
+     * @see #getShortVolumeFromLong and #getFeeFactor
+     * Detailed maths: https://github.com/scionaltera/arbitrader/issues/325
+     */
+    private BigDecimal getVolumeForLongTrade(BigDecimal longMaxExposure, BigDecimal shortMaxExposure, BigDecimal longPrice, BigDecimal shortPrice, BigDecimal longFee, BigDecimal shortFee) {
```

Comment:
Would it make sense to make a new object like `TradeVolumes` with two `BigDecimal`s in it so you could have a method `getTradeVolumes(...)` that just figures it all out in one shot and hands it to you?

I do like having the lower level methods available to get long from short and short from long, but in the higher level trade logic it might be nice to hide the details.

Reply:
I like the idea, a TradeVolumes could hold the 4 volumes (long, long adjusted for FeeComputation.CLIENT, short, short adjusted for FeeComputation.CLIENT) and hold the logic to adjust the volumes according to the fees/step size etc.